src/yukon/backend/src/devserv/description/example/example_sys_description.json

```
@@ -0,0 +1,152 @@
+                        },
+                        {
+                                "id": 3,
+                                "name": "esc_back_left",
+                                "health": "ERROR",
+                                "mode": "MAINTAINANCE",
+                                "uptime": 100,
+                                "vendorCode": 32893
+                        },
+                        {
+                                "id": 4,
+                                "name": "esc_back_right",
+                                "health": "WARNING",
+                                "mode": "SOFTWARE_UPDATE",
+                                "uptime": 100,
+                                "vendorCode": 62342
+                        },
+                        {
+                                "id": 5,
+                                "name": "ms5525_airspeed",
+                                "health": "OK",
+                                "mode": "OFFLINE",
+                                "uptime": 456,
+                                "vendorCode": 8678
+                        },
+                        {
+                                "id": 6,
+                                "name": "servo_right_elevon",
+                                "health": "OK",
+                                "mode": "OPERATIONAL",
+                                "uptime": 45,
+                                "vendorCode": 13434
+                        },
+                        {
+                                "id": 7,
+                                "name": "servo_left_elevon",
+                                "health": "OK",
+                                "mode": "MAINTAINANCE",
+                                "uptime": 456,
+                                "vendorCode": 21431
+                        }
+                ],
+                "grv": [{
+                                "nodeId": 1,
+                                "nodeName": "esc_front_left",
+                                "registerName": "drone.param.naming",
+                                "value": {
+                                        "_type_": "uavcan.register.Access.Request.1.0",
+                                        "foo": {
+                                                "_type_": "uavcan.register.Value.1.0",
+                                                "first": "test1",
+                                                "last": "test2"
+                                        },
+                                        "bar": 123
+                                },
+                                "mutable": true,
+                                "persistent": true
+                        },
+                        {
+                                "nodeId": 2,
+                                "nodeName": "esc_front_right",
+                                "registerName": "drone.pub.esc_status",
```

```suggestion
                                "registerName": "uavcan.pub.esc_status",
```

Did you mean `uavcan`?

---

No. It's actually `drone.`, since it's the namespace we are considering for the data types.

---

But this register contains the subject-ID for the ESC status subject, right? If my understanding is correct, then it should be `uavcan.`. The namespace name for the DS-015 types bears no relevance here because 1. the register names are standardized as described in the uavcan.register.Access type; 2. the namespace for the DS-015 will be not `drone` but `regulated.drone`.

---

@pavel-kirienko:
1. where exactly is that standardised in https://github.com/UAVCAN/public_regulated_data_types/blob/master/uavcan/register/384.Access.1.0.uavcan? I don't read anywhere that it has to start with `uavcan.`. The table that you show there is not clear, because `uavcan.pub.esc_status` is not the same as `"uavcan.<node_id>.esc_status"`.
2. That will only happen if the `drone` profile (DS-015) messages end-up being added into the UAVCAN public regulated data types. Our data types will need an abstraction on the DSDL's that allow to establish a range of port identifiers for the functions. But let's talk about it offline, since this subject needs to be clarified between the parties.

I will change this to `uavcan` anyway to avoid confusion.

---

1. It is not stated explicitly that reserved name patters start with `uavcan.`, but it stands to reason that a generic reserved pattern cannot start with `drone.`. I wrote about it in https://forum.uavcan.org/t/automatic-configuration-of-port-identifiers/840.

2. Not having the drone standard types in the regulated namespace defeats at least part of its purpose.